services/replicator/replicator.go

```diff
@@ -1102,6 +1106,36 @@ func (r *Replicator) SetAckOffset(ctx thrift.Context, request *shared.SetAckOffs
 	})
 	r.m3Client.IncCounter(metrics.ReplicatorSetAckOffsetScope, metrics.ReplicatorRequests)
 
+	var cgExtentCreated bool
+	r.knownCgExtentsMutex.RLock()
+	_, cgExtentCreated = r.knownCgExtents[request.GetExtentUUID()]
+	r.knownCgExtentsMutex.RUnlock()
+
+	if !cgExtentCreated {
+		// make sure the cg extent is created locally before accepting the SetAckOffset call.
+		// otherwise SetAckOffset will create the cg extent entry with no store uuid or output host uuid
+		// and we may not be able to clean up the entry eventually.
+		extent, err := r.metaClient.ReadConsumerGroupExtent(nil, &metadata.ReadConsumerGroupExtentRequest{
+			ConsumerGroupUUID: common.StringPtr(request.GetConsumerGroupUUID()),
+			ExtentUUID:        common.StringPtr(request.GetExtentUUID()),
+		})
+		if err != nil {
+			lcllg.WithField(common.TagErr, err).Error(`SetAckOffset: Failed to read cg extent locally`)
+			r.m3Client.IncCounter(metrics.ReplicatorSetAckOffsetScope, metrics.ReplicatorFailures)
+			return err
+		}
+		if len(extent.GetExtent().GetStoreUUIDs()) < 1 {
```

Comment:
"ReadConsumerGroupExtent" (in the metadata client), already checks for this condition .. and returns an error. So the check here is redundant .. and will likely never be reached.